src/logic/calculate.js

```diff
@@ -0,0 +1,107 @@
+import operate from './operate';
+import formatNumber from '../helpers/common';
+
+const calculate = (calculator, btn) => {
+  let {
+    total, next, operator, prev,
+  } = calculator;
+
+  const numbers = [
+    '9', '8', '7',
+    '6', '5', '4',
+    '3', '2', '1',
+    '0',
+  ];
+
+  const operators = [
+    '/', '*', '+', '-',
+  ];
+
+  if (numbers.includes(btn)) {
+    total = null;
+    if (!prev || prev === '0') {
+      prev = btn;
+    } else if (prev && !operator) {
+      prev += btn;
+    } else if ((operator && !next) || (next && next[0] === '0')) {
+      next = btn;
+    } else if (operator && next) {
+      next += btn;
+    }
+  }
+
+  if (operators.includes(btn)) {
+    if ((!prev || prev === '0') && total) {
+      prev = total;
+      total = null;
+    } else if (next) {
+      prev = operate(prev, next, operator).toString();
+      next = '0';
+      total = null;
+    }
+    operator = btn;
+  }
+
+  if (btn === '%') {
+    if (prev && !operator) {
+      total = (prev / 100).toString();
+      prev = '0';
+    }
+    if (next) {
+      next = (next / 100).toString();
+    }
+  }
```

Comment:
- [ ] In this case, you should use an operation from the `operate` function. Create a conditional there for this case and call the `modifier` function from the `Big.js` library in order to convert the number to a percentage 😉

Reply:
Fixed. Thanks for the review